src/anonymeter/evaluators/singling_out_evaluator.py

```diff
@@ -408,6 +425,13 @@ class SinglingOutEvaluator:
     control : pd.DataFrame (optional)
         Independent sample of original records **not** used to create the
         synthetic dataset. This is used to evaluate the excess privacy risk.
+    max_attempts : int or None, default is None
+        Maximum number of attempts that the attacker can make to generate
+        a singling out query. This is useful to avoid excessively long running
+        calculations which can happen if the tasks of generating enough singling
+        out queries is too hard. If ``max_attempts`` is None, no limit will be
+        imposed.
```

Comment:
nit: I'd give an explanation in what cases the generation might take took long. Also, to me it sounds like it might be a good idea to limit the `max_attemts` anyway by assigning some rather high value. We'd like to avoid users running infinite computations :) This way we'll also have one less optional in the code.

Reply:
I also thought about setting `max_attempts` to a high number (say, 10^6)... yes, let's do that. I'd keep the optional anyhow, so that it's always possible to just keep running until the attacker is satisfied. Or do you prefer `max_attempts=-1` in that case? I think `max_attempts=None` is clearer. 

About explaining when the generation would take too long, it's, as you know, pretty difficult, since depending on the  hyperparameters and the datasets there are potentially many ways to get stuck.

Reply:
I'm fine keeping optional in this case.

Reply:
so no 10^6 as default? I can put it there and modify the warning message that comes when the limit is reached to mention the possibility of disabling the check. WDYT?